packages/dd-trace/test/config.spec.js

```diff
@@ -2372,4 +2372,89 @@ describe('Config', () => {
       expect(taggingConfig).to.have.property('maxDepth', 7)
     })
   })
+
+  context('library config', () => {
```

Comment:
I feel like there are a few tests missing here, for example the priority order of the new sources with the old sources, error cases, etc. Only 3 tests seems a little small for the amount of code added.

Reply:
There is already a test about config priority, I have added more sources. I have also added the following tests:
- Invalid yaml
- Unknown key

See [add tests](https://github.com/DataDog/dd-trace-js/pull/5126/commits/343c9c2bc6af5c7bacc98623277039c3b55f4c72)